plugin/pkg/scheduler/factory/factory.go

```diff
@@ -991,3 +992,27 @@ func (p *podConditionUpdater) Update(pod *v1.Pod, condition *v1.PodCondition) er
 	}
 	return nil
 }
+
+type podPreemptor struct {
+	Client clientset.Interface
+}
+
+func (p *podPreemptor) GetUpdatedPod(pod *v1.Pod) (*v1.Pod, error) {
+	return p.Client.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{})
+}
+
+func (p *podPreemptor) PreemptPod(pod *v1.Pod) error {
+	return p.Client.CoreV1().Pods(pod.Namespace).Delete(pod.Name, &metav1.DeleteOptions{})
```

Comment:
preemption doesn't use pods/evict?

Reply:
No. There was a discussion on this in the past. The final outcome is that scheduler will not use eviction as it may need to override pod disruption budget.